tests/tls.go

```diff
@@ -118,6 +133,89 @@ func (s *TLSFunctionalSuite) TestHTTPMTLS() {
 	s.Require().Equal(tlsCertCommonName, authInfo.(*authorization.AuthInfo).TLSSubject.CommonName)
 }
 
+func (s *TLSFunctionalSuite) TestSameClusterCallbackMTLS() {
```

Comment:
nit: you don't even have to test this with MTLS, you can instead put an invalid host in the callback template DC and put this test in `nexus_workflow_test.go`.